File: Sequence/q1-q99/q70-q79/q75/q75.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        if (nums.size() < 2)
            return;
        int left = 0, mid = 0, right = nums.size() - 1;
        while (mid <= right) {
            if (nums[mid] == 0) {
                swap(nums[left++], nums[mid++]);
            } else if (nums[mid] == 1) {
                ++mid;
            } else {
                swap(nums[mid], nums[right--]);
            }
        }
            
    }
};

File: Sequence/q1-q99/q70-q79/q75/q75.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        left, middle, right = 0, 0, len(nums) - 1
        while middle <= right:
            if nums[middle] == 0:
                temp = nums[middle]
                nums[middle] = nums[left]
                nums[left] = temp
                middle += 1
                left += 1
            elif nums[middle] == 1:
                middle += 1
            else:
                temp = nums[middle]
                nums[middle] = nums[right]
                nums[right] = temp
                right -= 1
                
                
